src/allotropy/parsers/bmg_mars/bmg_mars_parser.py

```diff
@@ -0,0 +1,270 @@
+        lines = read_to_lines(named_file_contents)
+        read_type = self._get_read_type(lines)
+
+        reader = LinesReader(lines)
+        raw_header = list(reader.pop_until("^,?Raw Data"))
+        header = Header.create(raw_header)
+
+        csv_data = list(reader.pop_until_empty())
+        wavelength = Wavelength.create(csv_data)
+        plate_well_count = get_plate_well_count(csv_data)
+        data = get_plate_data(csv_data)
+
+        return self._get_model(
+            header,
+            data,
+            filename,
+            read_type,
+            wavelength,
+            plate_well_count,
+        )
+
+    def _get_model(
+        self,
+        header: Header,
+        data: pd.DataFrame,
+        filename: str,
+        read_type: ReadType,
+        wavelength: Wavelength,
+        plate_well_count: TQuantityValueNumber,
+    ) -> Model:
+
+        return Model(
+            plate_reader_aggregate_document=PlateReaderAggregateDocument(
+                plate_reader_document=self._get_plate_reader_document(
+                    data, read_type, header, wavelength, plate_well_count
+                ),
+                device_system_document=DeviceSystemDocument(
+                    device_identifier="N/A",
+                    model_number="N/A",
+                    product_manufacturer="BMG LABTECH",
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=filename,
+                    UNC_path=header.path,
+                    software_name="BMG MARS",
+                    ASM_converter_name=self.get_asm_converter_name(),
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+            ),
+            field_asm_manifest="http://purl.allotrope.org/manifests/plate-reader/BENCHLING/2023/09/plate-reader.manifest",
+        )
+
+    def _get_read_type(self, lines: list[str]) -> ReadType:
+        read_type = assert_not_none(
+            re.search(RE_READ_TYPE, "\n".join(lines), flags=re.MULTILINE),
+            msg="Read type not found.",
+        ).group(0)
+        patterns = {
+            "Absorbance": ReadType.ABSORBANCE,
+            "Fluorescence": ReadType.FLUORESCENCE,
+        }
+
+        for key in patterns:
+            if key in read_type:
+                read_type_key = patterns[key]
+        return read_type_key
+
+    def _get_device_control_aggregate_document(
+        self,
+        wavelength: Wavelength,
+        read_type: ReadType,
+    ) -> DeviceControlAggregateDocument:
+
+        if read_type == ReadType.ABSORBANCE:
+            return UltravioletAbsorbancePointDetectionDeviceControlAggregateDocument(
+                device_control_document=[
+                    UltravioletAbsorbancePointDetectionDeviceControlDocumentItem(
+                        device_type="absorbance detector",
+                        detection_type="absorbance",
+                        detector_wavelength_setting=safe_value(
```

Comment:
A bit surprised by the checks here -- seems like `wavelength` must be set, so the `if wavelength.wavelength` check would always pass unless wavelength is 0 or an empty string, are those conditions we'd expect to see in a data file? If not, seems like this could just be `TQuantityValueNanometer(wavelength.wavelength)`, and if so, I think it'd be better to check for that when creating the `Wavelength` object.

Reply:
Simplified detector_wavelength_setting for UV and Fluorescence